csrc/scheduler/vectorize_helper.cpp

```diff
@@ -933,12 +933,11 @@ int64_t getVectorizationFactorTransposeGroup(
   std::vector<IterDomain*> virtual_innermost_dim;
   // find the virtual_innermost_dim in reference so we can later map
   // that to individual TensorView in vec_tv.
-  for (const auto& dim : dims_to_merge) {
+  for (const auto dim : dims_to_merge) {
     virtual_innermost_dim.insert(
-        virtual_innermost_dim.begin(), reference->axis(static_cast<int>(dim)));
+        virtual_innermost_dim.begin(), reference->axis(dim));
   }
-  virtual_innermost_dim.push_back(
-      reference->getLogicalDomain()[inner_most_dim]);
+  virtual_innermost_dim.push_back(reference->axis(inner_most_dim));
```

Comment:
Just in case:
```suggestion
 // Make sure it is not yet parallelized
  NVF_ERROR(reference->axis(inner_most_dim)->getParallelType() == ParallelType::Serial);
  virtual_innermost_dim.push_back(reference->axis(inner_most_dim));
```